### PR TITLE
deploy/helm-chart: remove openTelemetry.enabled option

### PR DESCRIPTION
**Pull request by paulfantom**
Signed-off-by: Paweł Krupa (paulfantom) <pawel@krupa.net.pl>

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue._

As in title. Fixes https://github.com/timescale/promscale/issues/1241

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [x] Updated the relevant documentation

### CODE REVIEW

**Review by cevian — CHANGES_REQUESTED** [review]
Wait why are we removing the option instead of changing the default from `false` to `true`. Shouldn't we still allow users to turn Otel tracing off?

**Comment by paulfantom** [comment]
@cevian That's what is described in issue - https://github.com/timescale/promscale/issues/1241.

**Comment by paulfantom** [comment]
Pinging @VineethReddy02 and @ramonguiu for clarification if we want to allow disabling tracing (as suggested by @cevian) or do we want to remove this option and always run with tracing enabled (as described in #1241).

File: deploy/helm-chart/values.yaml [hunk 1]
```diff
@@ -28,8 +28,6 @@ extraEnv: []
 # Currently OpenTelemetry tracing support is in beta
 # so tracing features needs to be explicitly enabled.
```
Comment (JamesGuthrie) [inline deploy/helm-chart/values.yaml:29]: These lines should also go.

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]

File: deploy/helm-chart/values.yaml (hunk 1 above)
Comment (paulfantom) [reply deploy/helm-chart/values.yaml]: fixed.

File: deploy/helm-chart/templates/deployment-promscale.yaml [hunk 2]
```diff
@@ -39,16 +39,10 @@ spec:
         - image: {{ .Values.image }}
           imagePullPolicy: {{ .Values.imagePullPolicy }}
           name: promscale-connector
-          {{- if or (.Values.openTelemetry.enabled) (.Values.extraArgs) }}
```
Comment (JamesGuthrie) [inline deploy/helm-chart/templates/deployment-promscale.yaml:42]: Do we not still need the check for `.Values.extraArgs`? I assume that if there are none set, the resulting args block is empty.

**Review by JamesGuthrie — APPROVED** [review]
LGTM, aside from the open question about whether it should still be configurable or not.

**Review by antekresic — APPROVED** [review]
👍

**Comment by VineethReddy02** [comment]
@paulfantom Offering an option to disable tracing would be ideal as users might prefer closing the Otel port if they are not using OpenTelmetry.

**Review by cevian — APPROVED** [review]
Ok discussed with @VineethReddy02  and all the disabling stuff needs to be reworked almost completely anyway. So removing this option completely from the helm chart (as this PR does) is the right thing to do right now.

**Comment by cevian** [comment]
To be clear: there is no need to change the PR, it should be merged as-is.

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**